### ISSUE TITLE
Specific parser for rules LHS

### ISSUE THREAD

**Issue by steinerkelvin** [Kindelia/Kindelia#124]
Current parsing for rules LHS is using is using `read_term` which:

1. is too flexible
2. don't allow big Names

Rules LHS shoud have an specific parser syntatically allowing only Ctrs, Nums, Vars, etc and also allowing big function Names.

**Comment by racs4** [comment]
I made a draft [here](https://github.com/racs4/Kindelia/blob/parser-lhs/src/hvm.rs#L332) where I transformed the lhs `Rule` attribute in a `Vec<Term>` instead of a `Term`. This means that the lhs is no more a `Term::Fun` and only store its arguments. Such change remove some checking `if let Term::fun{..} = lhs` in some places and removes the predicted error with "big names". 
But, while I was making the alterations, I perceived that codes like
```
fun (Test) {
  (Testa #2) = #4
  (Testa #3) = #6
}

run {
  (Done (Test #2))
}
```
runs perfectly (it already worked like this before my alteration). I can check this in the parser and raise an error or we can change the way functions are written:
```
fun (Test) {
  (#2) = #4
  (#3) = #6
}

run {
  (Done (Test #2))
}
```
